### DISCUSSION TITLE
transistionProprety on height not working

### DISCUSSION THREAD

**Discussion by MatthiasScreed** [tailwindlabs/tailwindcss#5715] [Help] [answered]
i try to create a transition-heigt on a group-hover  so i put this in my tailwindconfigs :

`module.exports = {
  purge: [
    './src/**/*.html',
    './src/**/*.js',
  ],
  darkMode: false, // or 'media' or 'class'
  theme: {
    extend: {},
  },
  variants: {
    display: ['hover', 'focus', 'group-hover'],
    extend: {
      transitionProperty: {
        'height': 'height'
      },
      scale: ['active', 'group-hover'],
    },
  },
  plugins: [],
}`

the code look like this : 
`<button class="flex items-center justify-end w-full group">
							<span>
								masquer
							</span>
							<span class=""> 
								<svg
								class="w-4 h-4 transition duration-150 ease-in-out transform fill-current group-hover:-rotate-180"
								xmlns="http://www.w3.org/2000/svg"
								viewBox="0 0 20 20"
								  >
									<path
										  d="M9.293 12.95l.707.707L15.657 8l-1.414-1.414L10 10.828 5.757 6.586 4.343 8z"
									/>
								  </svg>
							</span>
						</button>
						<div class="invisible w-full h-0 px-3 py-2 duration-500 ease-in-out transform border-b transition-height group-hover:h-full group-hover:visible">
							<div class="flex">
								<img class="inline-block px-4 py-2 border-r-4" src="https://www.guideduportage.com/images/bannieres//10.gif" alt="">
								<ul class="ml-4">
									<li>https://www.cetelem.fr</li>
								</ul>
							</div>
							<div class="flex items-center justify-end space-x-3">
								<a href="" class="inline-block capitalize">supprimer</a>
								<a href="" class="inline-block px-3 py-2 font-bold text-white uppercase bg-green-500">editer</a>
							</div>
						</div>`
but the group-hover doesn't working  i don't know why

**Comment by simonswiss** [comment, marked as answer]
Hey!

There are a few things wrong with your code, and I have tried to put together a Tailwind Play that fixes them:

https://play.tailwindcss.com/BYqEOdPoSx?file=config

---

Here are some of the problems based on your original code:

1. Your `transitionProperty` should extend the `theme`, not the `variants`.

```diff
module.exports = {
  theme: {
    extend: {
+      transitionProperty: { height: 'height' },
    },
  },
variants: {
  display: ['hover', 'focus', 'group-hover'],
  extend: {
-      transitionProperty: { height: 'height' },
    scale: ['active', 'group-hover'],
  },
},
  plugins: [],
}
```

2. You also need to extend the `rotate`, `visibility` and `height` variants to add `group-hover`:

```diff
variants: {
  display: ['hover', 'focus', 'group-hover'],
  extend: {
    scale: ['active', 'group-hover'],
+    rotate: ['group-hover'],
+    visibility: ['group-hover'],
+    height: ['group-hover'],
  },
},
```

3. currently your invisible `div` tries to use `group-hover`, but is not nested inside an element with the class of `group`, so it's not going to do anything. The `group-hover:` expects a parent element to have a class of `group`, and will be applied when _that_ element is hovered.

Now, for the point 2, I highly recommend you start using Tailwind's `JIT` mode, so you can completely forget about extending variants, they'll all work out of the box:

https://tailwindcss.com/docs/just-in-time-mode

---

Besides all that: as annoying as it is, you cannot animate a height from `0px` to `auto` or `100%`, because the browser needs to know the total height of the element to calculate that transition. You'd typically need a bit of JavaScript to calculate the element's full height, and then animate towards that.

In my example, I used `h-64` to show how it works, while `h-full` does not.

Hope it helps you make steps towards a solution 👍

**Reply by MatthiasScreed** [reply]
thanks you very much work perfectly

**Reply by simonswiss** [reply]
Glad to hear this was helpful 🙌
